src/grafana_agent.py

```diff
@@ -1066,6 +1067,21 @@ def _instance_name(self) -> str:
 
         return socket.getfqdn()
 
+    @property
+    def log_level(self) -> str:
+        """The log level configured for the charm."""
+        # Valid upstream log levels in server_config
+        # https://grafana.com/docs/agent/latest/static/configuration/server-config/#server_config
+        allowed_log_levels = ["debug", "info", "warn", "error"]
+        log_level = cast(str, self.config.get("log_level")).lower()
+
+        if log_level not in allowed_log_levels:
+            logging.warning(
+                f'Invalid loglevel: {log_level} given, {"/".join(allowed_log_levels)} allowed. defaulting to INFO loglevel.'
+            )
+            log_level = "info"
```

Comment:
Can we replace the string literal with what's in e.g. ops.meta?

Reply:
[ops.meta](https://ops.readthedocs.io/en/latest/#ops.CharmMeta) does not offer config options. Is this what you meant?